juju/juju
# Multiple bundle config

When deploying complex models, often the bundle is broken up into multiple components.

This branch also fixes the problem with YAML anchors used on the bundle config.

## QA steps

Create two files:
remove-easyrsa.yaml (http://pastebin.ubuntu.com/25929417/)
placement.yaml (http://pastebin.ubuntu.com/25929422/)
Then run the following in an empty model:
$ juju deploy canonical-kubernetes --bundle-config ~/canonical/remove-easyrsa.yaml --bundle-config ~/canonical/placement.yaml --dry-run

The output should match http://pastebin.ubuntu.com/25929429/ where there is no easyrsa deployed, and all the units are colocated on three machines.

## Documentation changes

The bundle documentation will need to be updated to outline the precedence rules for the config with respect to the machines, applications, and series. It should also explain how to use config to remove an application from a bundle, as well as how to use YAML anchors for common config.

## Bug reference

https://bugs.launchpad.net/juju/+bug/1721603
https://bugs.launchpad.net/juju/+bug/1728593
